Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -18,31 +18,80 @@
 public interface SeekableChannelsProvider extends SafeCloseable {
 
     /**
-     * Take the file source path or URI and convert it to a URI object.
+     * Take the file source path or URI string and convert it to a URI object.
      *
      * @param source The file source path or URI
+     * @param isDirectory Whether the source is a directory
      * @return The URI object
      */
-    static URI convertToURI(final String source) {
+    static URI convertToURI(final String source, final boolean isDirectory) {
```

Comment:
Is this class the right place for the utility methods?

Reply:
Looks like there is a Deephaven core uri library and package `io.deephaven.uri` with a public class `UriHelper`, I can move these methods there. I would need to add a new dependency though for this.

Reply:
Moved the methods to FileUtils, let me know how this looks.

Reply:
This is a good place. Mention this in a team channel, it's educational and we should share the knowledge.